### DISCUSSION TITLE
Welcome to Tensorflow-2-Reinforcement-Learning-Cookbook Discussions!

### DISCUSSION THREAD

**Discussion by praveen-palanisamy** [PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook#1] [General]
## 👋 Welcome to the TFRL Cookbook Community!
  This is a place where you can connect with the author, other readers and audience for the ["TensorFlow 2 Reinforcement Learning Cookbook" ](https://www.packtpub.com/product/tensorflow-2-reinforcement-learning-cookbook/9781838982546) and [code](https://github.com/PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook/):
  * Ask any questions related to the book or code that you’re wondering about.
  * Showcase your own creations and customization.
  * Share your thoughts on how you are using the book/code with others.
  * Welcome others and be kind :). Remember that this is a community we build together 💪.

  You can get started by commenting below with an introduction of yourself and tell us what you are working on.

**Comment by ghost** [comment]
Chapter 1 example problem MountainCarContinous environment and the DiscretePolicy example code throw the following error:

ValueError: in converted code:

    <ipython-input-18-6316ede25ee8>:12 call  *
        return self.logits(self.dense1(x))
    C:\Users\billm\anaconda3\envs\tfrl-cookbook\lib\site-packages\tensorflow_core\python\keras\engine\base_layer.py:748 __call__
        self._maybe_build(inputs)
    C:\Users\billm\anaconda3\envs\tfrl-cookbook\lib\site-packages\tensorflow_core\python\keras\engine\base_layer.py:2116 _maybe_build
        self.build(input_shapes)
    C:\Users\billm\anaconda3\envs\tfrl-cookbook\lib\site-packages\tensorflow_core\python\keras\layers\core.py:1101 build
        raise ValueError('The last dimension of the inputs to `Dense` '

    ValueError: The last dimension of the inputs to `Dense` should be defined. Found `None`.

I double checked and typed the code per the book examples - seems to be in class Brain call function. Any idea?

**Reply by praveen-palanisamy** [reply]
This is the related-to-the/same discussion we are having at: https://github.com/PacktPublishing/Tensorflow-2-Reinforcement-Learning-Cookbook/discussions/6
Let's use dedicated issue/discussion item for individual issues and use this general channel for general topics. Thanks!
